### ISSUE TITLE
Support for sACN

### ISSUE THREAD

**Issue by SammyTheHand** [cbdevnet/midimonster#9]
Would it be possible that this could support Streaming ACN - E1.31?

**Comment by cbdevnet** [comment]
Yes, definitely!
An sACN backend would probably share a lot of architecture with the existing ArtNet backend, so it might not even be that hard!

If you'd like to try your hand at it:

* I found a [protocol spec](http://tsp.esta.org/tsp/documents/docs/E1-31-2016.pdf) document for sACN/ANSI E1.31-2016
* The backend lifecycle is documented in the big comment block in midimonster.h
* The ArtNet backend as well as (in parts) the OSC backend could probably serve as examples for implementing it

I'm currently busy with exam preparation until Thursday, but after that I might be able to sneak in some hours of implementing this :)

Thanks for the suggestion!

**Comment by dmeltzer** [comment]
I'm also very interested in this.  Might poke at it over the weekend.

**Comment by dmeltzer** [comment]
The only open source one I'm aware of is OLA.  I've not played enough with it to ever make it work though.  https://github.com/OpenLightingProject/ola/tree/master/libs/acn

**Comment by cbdevnet** [comment]
Just remembered that the MAdot2 series grant one Universe without having MA hardware connected, and the OnPC seems to support sACN, too (at least according to their docs) - might try with that

**Comment by SammyTheHand** [comment]
Test with sACN Viewer: http://docsteer.github.io/sacnview/

**Comment by cbdevnet** [comment]
I've just pushed a partial sACN backend implementation, currently only supporting input via sACN.
The mapping syntax is the same as the ArtNet backend, documentation will be updated when output is implemented

**Comment by dmeltzer** [comment]
I'll give this a try with eos tomorrow, my current project would be much better off with sACN instead of artnet.

**Comment by cbdevnet** [comment]
Output support is now implemented, still missing are

* Keepalive frames (same as the ArtNet backend, and will probably fix it for both then)
* Universe discovery
* Documentation
* A way to designate an instance as output-only

**Comment by dmeltzer** [comment]
Thanks for this.  I gave sACN support a try today and it seemed to work fairly well out of the box.  I noticed that it used universe where artnet used uni, which was not immediately clear in the configs (or maybe you've updated artnet as well).

I did notice more glitches/dropped messages today than I did when testing with artnet last week.  I have not dug into the details enough to see if it's midimonster dropping messages or if its something else in my network thats changed, just an observation.
